### DISCUSSION TITLE
Query invalidation and refetch

### DISCUSSION THREAD

**Discussion by vjau** [TanStack/query#7053] [Q&A]
I have been using `Query` for some time now, so i thought i had a clear grasp of how it was working, but suddenly, some piece of my code is not working as intended, so i have double thoughts. 
I am using a query in a component with `useQuery`, showing a list of things.
When committing a change (actually a deletion of one element in the list) with a mutation, i invalidate the query in an `onSuccess` handler.
It has always been my understanding that the query should then refetch to display the new list, however that is not what happens. My query is _not_ refetching (triple checked with various means, one being the query devtools) and i must force the component to recreate to get updated data.
When using `refetch` instead of `invalidate`, my query refetch and the updated data correctly shows. 
I suspect this is a bug with the trpc adapter, but i would like to be sure that my understanding of the intended behavior of `Query` is correct. 
Thank you.

**Comment by TkDodo** [comment]
most often, it just means that the keys aren't matching / you're not invalidating the correct query

**Reply by vjau** [reply]
The thing is that trpc add a feature to invalidate all query by root.
Eg, if you hava a query `trpc.foo.bar.getBaz` and a query `trpc.foo.bar.getBap` you can invalidate at the level you want, `trpc.foo.bar.invalidate()` or even `trpc.foo.invalidate()`, but i suspect there is a bug somewhere in their wrapper. 
I will investigate on their side, thank you.
